flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/stream/ValuesTestPrograms.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.planner.plan.nodes.exec.stream;
+
+import org.apache.flink.table.test.program.SinkTestStep;
+import org.apache.flink.table.test.program.TableTestProgram;
+
+/** {@link TableTestProgram} definitions for testing {@link StreamExecValues}. */
+public class ValuesTestPrograms {
+
+    static final TableTestProgram VALUES_TEST =
+            TableTestProgram.of("values-test", "validates values node")
+                    .setupTableSink(
+                            SinkTestStep.newBuilder("sink_t")
+                                    .addSchema("b INT", "a INT", "c VARCHAR")
+                                    .consumedBeforeRestore("+I[1, 2, Hi]", "+I[3, 4, Hello]")
```

Comment:
Is that an expected behaviour? This suggests `values` does not support properly taking a savepoint, which I believe is not the case. 

Can we test the `Values` in a different way? I guess the `RestoreTestBase` does not fit well here.

Reply:
Changed the test to be a non restore test. Also this test does not depend on a savepoint. It only takes a json compiled plan and executes it.